### PR TITLE
Ignore keras test if environment is not ready and add test method for loss

### PR DESCRIPTION
**Pull request by zhichao-li**
## What changes were proposed in this pull request?

Ignore keras test if environment is not ready and add test method for loss

## How was this patch tested?
unittest

### CODE REVIEW

**Comment by zhichao-li** [comment]
cc @yiheng  @wzhongyuan

File: spark/dl/src/test/scala/com/intel/analytics/bigdl/keras/KerasBaseSpec.scala [hunk 1]
```diff
@@ -14,11 +14,28 @@
...
+import com.intel.analytics.bigdl.nn.abstractnn.{AbstractCriterion, AbstractModule, Activity}
 import com.intel.analytics.bigdl.tensor.Tensor
-import org.scalatest.{BeforeAndAfter, FlatSpec, Matchers}
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric.NumericFloat
+import org.scalatest.{BeforeAndAfterEach, FlatSpec, Matchers}
 
-class KerasBaseSpec extends FlatSpec with BeforeAndAfter with Matchers {
+import scala.sys.process._
+
+class KerasBaseSpec extends FlatSpec with BeforeAndAfterEach with Matchers {
```
Comment (yiheng) [inline spark/dl/src/test/scala/com/intel/analytics/bigdl/keras/KerasBaseSpec.scala]: change to abstract? extend BigDLSpecHelper?

File: spark/dl/src/test/scala/com/intel/analytics/bigdl/keras/KerasBaseSpec.scala [hunk 2]
```diff
@@ -14,11 +14,28 @@
...
+    // Skip unitest if environment is not ready
+    try {
+    Seq("python", "-c", "import keras; import tensorflow").!!
+    } catch {
+      case e: Throwable => cancel("python or keras or tensorflow is not installed", e)
+    }
+  }
+
+  override def beforeEach() {
+    ifskipTest()
```
Comment (yiheng) [inline spark/dl/src/test/scala/com/intel/analytics/bigdl/keras/KerasBaseSpec.scala]: will this be run in each test. I remember if you call cancel before test case, the test will be marked as failure. It is marked as skipped when you call cancel in each test.

File: spark/dl/src/test/scala/com/intel/analytics/bigdl/keras/KerasBaseSpec.scala [hunk 3]
```diff
@@ -57,5 +74,20 @@ class KerasBaseSpec extends FlatSpec with BeforeAndAfter with Matchers {
       }
     }
   }
+
+  def checkOutputAndGradForLoss(bmodel: AbstractCriterion[Tensor[Float], Tensor[Float], Float],
```
Comment (yiheng) [inline spark/dl/src/test/scala/com/intel/analytics/bigdl/keras/KerasBaseSpec.scala]: only loss? how can we test layer?

**Comment by yiheng** [comment]
Please check my comments.

File: spark/dl/src/test/scala/com/intel/analytics/bigdl/keras/KerasBaseSpec.scala (hunk 2 above)
Comment (zhichao-li) [reply spark/dl/src/test/scala/com/intel/analytics/bigdl/keras/KerasBaseSpec.scala]: oh, that's too bad which means we need to explicitly call this `ifskipTest` in each test.

Comment (zhichao-li) [reply spark/dl/src/test/scala/com/intel/analytics/bigdl/keras/KerasBaseSpec.scala]: would change it if there's no other better option.

Comment (yiheng) [reply spark/dl/src/test/scala/com/intel/analytics/bigdl/keras/KerasBaseSpec.scala]: verify it when run keras code?

**Comment by wzhongyuan** [comment]
Looks good, tested on top of it locally, will add my test once it's merged :)

**Pull request merged**